common/src/main/java/com/microsoft/identity/common/internal/logging/CommonCoreLogger.java

```diff
@@ -0,0 +1,264 @@
+        }
+
+        mExternalLogger.set(externalLogger);
+    }
+
+    /**
+     * Send a {@link LogLevel#ERROR} log message without PII.
+     */
+    public static void error(final String tag, final String correlationID, final String errorMessage,
+                             @Nullable ADALError errorCode, final Throwable exception) {
+        getInstance().log(tag, LogLevel.ERROR, correlationID, errorMessage, errorCode, exception, false);
+    }
+
+    /**
+     * Send a {@link LogLevel#ERROR} log message with PII.
+     */
+    public static void errorPII(final String tag, final String correlationID, final String errorMessage,
+                                @Nullable ADALError errorCode, final Throwable exception) {
+        getInstance().log(tag, LogLevel.ERROR, correlationID, errorMessage, errorCode, exception, true);
+    }
+
+    /**
+     * Send a {@link LogLevel#WARN} log message without PII.
+     */
+    public static void warn(final String tag, final String correlationID, final String message, @Nullable ADALError errorCode) {
+        getInstance().log(tag, LogLevel.WARN, correlationID, message, errorCode, null, false);
+    }
+
+    /**
+     * Send a {@link LogLevel#WARN} log message with PII.
+     */
+    public static void warnPII(final String tag, final String correlationID, final String message, @Nullable ADALError errorCode) {
+        getInstance().log(tag, LogLevel.WARN, correlationID, message, errorCode, null, true);
+    }
+
+    /**
+     * Send a {@link LogLevel#INFO} log message without PII.
+     */
+    public static void info(final String tag, final String correlationID, final String message, @Nullable ADALError errorCode) {
+        getInstance().log(tag, LogLevel.INFO, correlationID, message, errorCode, null, false);
+    }
+
+    /**
+     * Send a {@link LogLevel#INFO} log message with PII.
+     */
+    public static void infoPII(final String tag, final String correlationID, final String message, @Nullable ADALError errorCode) {
+        getInstance().log(tag, LogLevel.INFO, correlationID, message, errorCode, null, true);
+    }
+
+    /**
+     * Send a {@link LogLevel#VERBOSE} log message without PII.
+     */
+    public static void verbose(final String tag, final String correlationID, final String message, @Nullable ADALError errorCode) {
+        getInstance().log(tag, LogLevel.VERBOSE, correlationID, message, errorCode, null, false);
+    }
+
+    /**
+     * Send a {@link LogLevel#VERBOSE} log message with PII.
+     */
+    public static void verbosePII(final String tag, final String correlationID, final String message, @Nullable ADALError errorCode) {
+        getInstance().log(tag, LogLevel.VERBOSE, correlationID, message, errorCode, null, true);
+    }
+
+    /**
+     * TODO 1. Need to discuss on how to keep the correlationID. CorrelationID should be per request => need to sync with Telemetry implementation

```

Comment:
Is this question unresolved? Looks like having the correlation ids in the params should be sufficient for logger usage..... I guess this may get 'heavy' to lug around in the caching logic but I don't see a way around this at the moment......

Reply:
Yes, it is unresolved. Need you guys' opinions.